hashing/blake2b/blake2b.go

```diff
@@ -6,38 +6,49 @@ import (
 	"golang.org/x/crypto/blake2b"
 )
 
-var b2bEmptyHash []byte
-
 // Blake2b is a blake2b implementation of the hasher interface.
 type Blake2b struct {
-	HashSize int
+	HashSize  int
+	emptyHash []byte
```

Comment:
I think we could correct this implementation, like making a constructor for Blake2B with a given hashSize, not accepting hashSizes of 0 - as it does not have any value. GetHasher function implementation can be put in constructor, as well as the call for setEmpty.

Reply:
Indeed, using a constructor could be a better approach, however since the full hashing package needs some refactoring (use []byte as input data), fix some logic in all the other hashers, I think is better to do it then